fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -598,11 +610,21 @@ private ScoredPlan planFilterWithInJoin(@Nonnull PlanContext planContext, @Nonnu
         }
 
         Verify.verifyNotNull(bestPlan);
-        
-        final RecordQueryPlan wrapped = inExtractor.wrap(planContext.rankComparisons.wrap(bestPlan.getPlan(), bestPlan.includedRankComparisons, metaData));
-        final ScoredPlan scoredPlan = new ScoredPlan(bestPlan.score, wrapped);
+        return new PlanWithInExtractor(bestPlan, inExtractor);
```

Comment:
So `bestPlan` is in fact the best plan _underneath_ the future in plan (inJoin or inUnion) together with the `inExtractor` that was used to get to the best plan. Is the right?

Reply:
Maybe give this method a javadoc to reflect that.

Reply:
Yeah, that's right. I've added Javadoc on these two methods to hopefully make that clearer.